### PR TITLE
Adds dedicated entities for all features

### PR DESCRIPTION
**Pull request by dbuezas**

### CODE REVIEW

File: custom_components/dbuezas_eq3btsmart/climate.py [hunk 1]
```diff
@@ -110,50 +99,31 @@ def json_serial(obj):
...
         """Initialize the thermostat."""
-        # We want to avoid name clash with this module.
         self.hass = _hass
-        self._mac = _mac
         self._current_temperature = None
         # TODO: refactor the is_setting_temperature mess.
         self._is_setting_temperature = False
-        self._thermostat = eq3.Thermostat(_mac, _device_name, _hass, self._on_updated)
+        self._thermostat = _thermostat
+        self._thermostat.set_on_update(self._on_updated)
```
Comment (dbuezas) [inline custom_components/dbuezas_eq3btsmart/climate.py:126]: 😁

File: custom_components/dbuezas_eq3btsmart/button.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add sensors for passed config_entry in HA."""
+    eq3 = hass.data[DOMAIN][config_entry.entry_id]
+    _LOGGER.debug(
+        "[%s] button: async_setup_entry",
+        eq3.name,
+    )
+
+    new_devices = [FetchScheduleButton(eq3)]
```
[the remark this answers is not in the document]
Comment (dbuezas) [reply custom_components/dbuezas_eq3btsmart/button.py:29]: Oh, it did add them, they are just not inside the device
image

[the remark this answers is not in the document]
Comment (dbuezas) [reply custom_components/dbuezas_eq3btsmart/button.py:29]: image
I think this is it

[the remark this answers is not in the document]
Comment (lkempf) [reply custom_components/dbuezas_eq3btsmart/button.py:29]: You need to implement device_info in button.py as well. I think the only relevant attribute there should be identifier

[the remark this answers is not in the document]
Comment (dbuezas) [reply custom_components/dbuezas_eq3btsmart/button.py:29]: image

I don't know if it is HA or the fact that this is the only thing I've done in python, but I find custom integrations to be quite a frustrating labyrinth some times...

[the remark this answers is not in the document]
Comment (dbuezas) [reply custom_components/dbuezas_eq3btsmart/button.py:29]: You are right, I had done it but it didn't work alone, it also requires the unique_id attribute to be defined in the entity

**Comment by dbuezas** [comment]
Work in progress:
image

**Comment by dbuezas** [comment]
Work in progress:
image

**Pull request merged**